src/base/bittorrent/private/speedmonitor.cpp

```diff
@@ -27,16 +27,24 @@
  * exception statement from your version.
  */
 
-#include <QList>
 #include "speedmonitor.h"
 
 void SpeedMonitor::addSample(const SpeedSample &sample)
 {
-    m_speedSamples.push_back(sample);
+    static_assert(MAX_SAMPLES > 0, "MAX_SAMPLES should > 0");
+
     m_sum += sample;
-    if (m_speedSamples.size() > MAX_SAMPLES) {
-        m_sum -= m_speedSamples.front();
-        m_speedSamples.pop_front();
+    if (m_speedSamples.size() >= MAX_SAMPLES) {
```

Comment:
@sledgehammer999, `boost::circular_buffer` is used now for speed graphs and I think that it's very usefull `fixed-size buffer` as needed in this case too.